torch/optim/asgd.py

```diff
@@ -427,17 +420,13 @@ def _multi_tensor_asgd(
             torch._foreach_mul_(new_etas, lr)
             torch._foreach_copy_(grouped_etas, new_etas)
         else:
-            step = grouped_state_steps[0].item()
-            new_etas = []
-            new_mus = []
-
-            for i in range(len(grouped_mus)):
-                new_eta = _to_tensor(
-                    lr / (1 + lambd * lr * step**alpha), device=device
-                )
-                new_etas.append(new_eta)
-                new_mu = _to_tensor(1 / max(1, step - t0), device=device)
-                new_mus.append(new_mu)
-
+            new_etas = [
+                torch.as_tensor(lr / (1 + lambd * lr * step**alpha), device=device)
```

Comment:
Let's correct this to be (1 + lambd * lr * step) ^ alpha as that was what it was historically (I blamed back to https://github.com/pytorch/pytorch/commit/8bb1d067026ac380dbd76cab34c8c6ee859542ef)